src/rnnlm/rnnlm-utils-test.cc

```diff
@@ -0,0 +1,177 @@
+// rnnlm/rnnlm-utils-test.cc
+
+#include <math.h>
+#include "rnnlm/rnnlm-utils.h"
+
+namespace kaldi {
+namespace rnnlm {
+
+void PrepareVector(int n, int ones_size, std::set<int> &outputs_set, vector<BaseFloat> &selection_probs) {
+  BaseFloat prob = 0;
+  BaseFloat prob_sum = 0;
+  for (int i = 0; i < n; i++) {
+    prob = RandUniform();
+    prob_sum += prob;
+    selection_probs.push_back(prob);
+  }
+  for (int i = 0; i < n; i++) {
+    selection_probs[i] /= prob_sum;
+  }
+  for (int i = 0; i < ones_size; i++) {
+    outputs_set.insert(rand() % n);
+  }
+}
+
+void UnitTestNChooseKSamplingConvergence(int n, int k, int ones_size) {
+  std::set<int> outputs_set;
```

Comment:
do not call this outputs_set.

In my code it is named so because of its meaning in the nnet framework;

here call it something like "must_sample_set"